templates/connect-inject-deployment.yaml

```diff
@@ -19,7 +19,7 @@ metadata:
     heritage: {{ .Release.Service }}
     release: {{ .Release.Name }}
 spec:
-  replicas: 1
+  replicas: 3
```

Comment:
curious what our thoughts are on this? will 2 cause election issues?
wondering if we bound it to # of server replicas or something, in case this is a demo environment?

Reply:
2 should also be fine. I put up 3 as a shot in the dark. the number of instances here are irrelevant. If you have a single instance running, it will assume it is the leader as it does not use a "consensus" protocol to establish leadership.

Reply:
Ah, interesting!
Okay thanks for the clarification. I'm just trying to think of how this might look if we have a demo environment where we've got a single kind node or something and then three of these try to spin up? 2 even seems too high in that case, but would be ideal in any other case. thoughts?

Reply:
I think 2 makes more sense to me. Demo env is tough, it definitely doesn't need 2 but folks won't change this for prod either and we default to 3 servers for the same reason. We could handle demo env another way maybe via profiles.

Reply:
Yeah, we're definitely pretty inconsistent. Although, if I were to guess it seems that we're leaning towards making it prod-ready by default rather than demo by default because servers and all gateways start with multiple instances. The only components that don't are sync-catalog, connect-injector, and controller. Sync catalog and injector could not run more than one instance, and for the controller we could probably switch too especially if we're thinking to combine it with the endpoints controller.

I think it'd make sense to default to a more HA setting (with 2 instances).

Reply:
Agree! 2 sounds good then!